### PR TITLE
Abstract revisions to dedicated controller; only include revisioned fields

### LINKED ISSUE #869: Create WP_JSON_Revisions_Controller
Because the schema for Revisions will be quite different than Posts and Pages, we should just create a separate controller for it.

This way, we don't need to register `/wp/posts/<id>/revisions` and can instead use `/wp/revisions/?parent=<id>`

After #844

### PR DESCRIPTION
**Pull request by danielbachhuber**
Fixes #869, #855 
See #894, #314

### CODE REVIEW

**Comment by danielbachhuber** [comment]
@WP-API/amigos #reviewmerge

File: lib/endpoints/class-wp-json-revisions-controller.php [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+	public function get_item_permissions_check( $request ) {
+
+		$response = $this->get_items_permissions_check( $request );
+		if ( ! $response || is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		$post = get_post( $request['id'] );
+		if ( ! $post || 'revision' !== $post->post_type ) {
+			return new WP_Error( 'json_post_invalid_id', __( 'Invalid revision ID.' ), array( 'status' => 404 ) );
```
Comment (joehoyle) [inline lib/endpoints/class-wp-json-revisions-controller.php]: Elsewhere we have not thrown invalid errors in the permissions checks. We'd typically not return any error in the case of the post not existing, or being the wrong post type - and handle that check in the `get_item` callback it's self.

**Comment by joehoyle** [comment]
Just a concern around the error checking done in the permissions callbacks, as I'm not sure if that is the best place for those checks. Apart from that - I'm happy.

File: lib/endpoints/class-wp-json-revisions-controller.php (hunk 1 above)
Comment (danielbachhuber) [inline lib/endpoints/class-wp-json-revisions-controller.php]: d3ff8cd

**Pull request merged**